services/pull/pull.go

```diff
@@ -464,7 +463,13 @@ func AddTestPullRequestTask(opts TestPullRequestOptions) {
 			log.Error("Find pull requests [base_repo_id: %d, base_branch: %s]: %v", opts.RepoID, opts.Branch, err)
 			return
 		}
+		baseRepo, err := repo_model.GetRepositoryByID(ctx, opts.RepoID)
+		if err != nil {
+			log.Error("GetRepositoryByID: %v", err)
+			return
+		}
 		for _, pr := range prs {
+			pr.BaseRepo = baseRepo // avoid loading again // FIXME: why only here does so but the code above doesn't do so
```

Comment:
FIXME

Reply:
The first loop get all pull requests with the given head repository id, so we cannot reuse the base repository.